tests/network.md

```diff
@@ -165,42 +165,50 @@ Exception: Failure "Simulated error".
 Working with UDP and endpoints:
 
 ```ocaml
-let run_dgram addr ~net sw =
-  let e1 = `Udp (addr, 8081) in
-  let e2 = `Udp (addr, 8082) in
-  let listening_socket = Eio.Net.datagram_socket ~sw net e2 in
+let run_dgram addr fn ~net sw =
+  let server_addr = `Udp (addr, 8082) in
+  let listening_socket = Eio.Net.datagram_socket ~sw net server_addr in
   Fiber.both
     (fun () ->
       let buf = Cstruct.create 20 in
-      traceln "Waiting to receive data on %a" Eio.Net.Sockaddr.pp e2;
+      traceln "Waiting to receive data on %a" Eio.Net.Sockaddr.pp server_addr;
       let addr, recv = Eio.Net.recv listening_socket buf in
       traceln "Received message from %a: %s"
       Eio.Net.Sockaddr.pp addr
       (Cstruct.(to_string (sub buf 0 recv)))
     )
-    (fun () ->
-      let e = Eio.Net.datagram_socket ~sw net e1 in
-      traceln "Sending data from %a to %a" Eio.Net.Sockaddr.pp e1 Eio.Net.Sockaddr.pp e2;
-      Eio.Net.send e e2 (Cstruct.of_string "UDP Message"))
+    (fun () -> fn ~net sw server_addr)
 ```
 
 Handling one UDP packet using IPv4:
 
+<!-- $MDX non-deterministic=output -->
 ```ocaml
-# run (run_dgram Eio.Net.Ipaddr.V4.loopback);;
+# let addr = Eio.Net.Ipaddr.V4.loopback in
+  run @@ run_dgram addr (fun ~net sw server_addr ->
+    let e1 = `UdpV4 in
+    let e = Eio.Net.datagram_socket ~sw net e1 in
+    traceln "Sending data from %s to %a" "UdpV4" Eio.Net.Sockaddr.pp server_addr;
+    Eio.Net.send e server_addr (Cstruct.of_string "UDP Message"));;
 +Waiting to receive data on udp:127.0.0.1:8082
-+Sending data from udp:127.0.0.1:8081 to udp:127.0.0.1:8082
-+Received message from udp:127.0.0.1:8081: UDP Message
++Sending data from UdpV4 to udp:127.0.0.1:8082
++Received message from udp:127.0.0.1:34435: UDP Message
 - : unit = ()
 ```
 
 Handling one UDP packet using IPv6:
 
+<!-- $MDX non-deterministic=output -->
 ```ocaml
-# run (run_dgram Eio.Net.Ipaddr.V6.loopback);;
+# let addr = Eio.Net.Ipaddr.V6.loopback in
+  run @@ run_dgram addr (fun ~net sw server_addr ->
+    let e1 = `UdpV6 in
+    let e = Eio.Net.datagram_socket ~sw net e1 in
+    traceln "Sending data from %s to %a" "UdpV6" Eio.Net.Sockaddr.pp server_addr;
+    Eio.Net.send e server_addr (Cstruct.of_string "UDP Message"));;
 +Waiting to receive data on udp:[::1]:8082
-+Sending data from udp:[::1]:8081 to udp:[::1]:8082
-+Received message from udp:[::1]:8081: UDP Message
++Sending data from UdpV6 to udp:[::1]:8082
++Received message from udp:[::1]:56756: UDP Message
 - : unit = ()
```

Comment:
maybe don't print the random port and keep the deterministic output ?
I think it also makes sense to keep one test using the bound socket (Eio.Net.datagram_socket `Udp...)

Reply:
ack.

Reply:
Done.